### PR TITLE
fix(KFLUXBUGS-1156): only publish when publish flag is true

### PR DESCRIPTION
**Pull request by scoheb**
- we first need to inspect the publish-on-push flag before we actually try to publish the image.
- this gives Product Owners the ability to hide content.

### CODE REVIEW

File: tasks/publish-pyxis-repository/publish-pyxis-repository.yaml [hunk 1]
```diff
@@ -95,13 +95,24 @@ spec:
...
             if [ -z "$PYXIS_REPOSITORY_ID" ]; then
                 echo Error: Unable to get Container Repository object id from Pyxis
                 echo "Pyxis response for ${PYXIS_REGISTRY}/${PYXIS_REPOSITORY}:"
                 echo $PYXIS_REPOSITORY_JSON
                 exit 1
             fi
+
+            # verify that publish_on_push is set to true.
+            # otherwise, do not publish the image.
+            PYXIS_REPOSITORY_PUBLISH_ON_PUSH=$(jq -r '.publish_on_push // ""' <<< $PYXIS_REPOSITORY_JSON)
```
Comment (theflockers) [inline tasks/publish-pyxis-repository/publish-pyxis-repository.yaml]: Maybe it should default to `"false"` instead of `""`.

Comment (scoheb) [reply tasks/publish-pyxis-repository/publish-pyxis-repository.yaml]: fixed! thanks

**Comment by happybhati** [comment]
@scoheb the task version and README needs to be updated before we merge.

**Comment by scoheb** [comment]
> @scoheb the task version and README needs to be updated before we merge.

good catch! updated!

File: tasks/publish-pyxis-repository/publish-pyxis-repository.yaml [hunk 2]
```diff
@@ -4,7 +4,7 @@ kind: Task
 metadata:
   name: publish-pyxis-repository
   labels:
-    app.kubernetes.io/version: "0.2.0"
+    app.kubernetes.io/version: "0.3.0"
```
Comment (happybhati) [inline tasks/publish-pyxis-repository/publish-pyxis-repository.yaml]: @scoheb Sorry to bother you with this but as commit title says fix(), the version should be "0.2.1"

**Review by happybhati — DISMISSED** [review]

File: tasks/publish-pyxis-repository/publish-pyxis-repository.yaml (hunk 2 above)
Comment (scoheb) [reply tasks/publish-pyxis-repository/publish-pyxis-repository.yaml]: fixed!

**Review by happybhati — APPROVED** [review]

**Pull request merged**